srcs/runners_manager/runner/RunnerFactory.py

```diff
@@ -96,7 +97,13 @@ def delete_runner(self, runner: Runner):
             logger.info(f'APIException catch, when try to delete the runner: {str(runner)}')
 
     def generate_runner_name(self, vm_type: VmType):
+        from web import redis_database
         vm_type.tags.sort()
-        return self.runner_name_format.format(index=self.runner_counter,
+        name = self.runner_name_format.format(index=self.runner_counter,
                                               organization=self.github_organization,
                                               tags='-'.join(vm_type.tags))
+        self.runner_counter += 1
+        # Check that a virtual machine hasn't this name already
+        if redis_database.redis.get(f'runners:{name}'):
+            return self.generate_runner_name(vm_type)
+        return name
```

Comment:
I don't get it here, it feels like we're gonna loop forever, what am I missing? Coffee :coffee: ? :smile:

Reply:
This fonction only generate the name, this if just check that we are not using the same name than an existing runner set in the redis database

Reply:
right, but as we are calling it recursively, won't we always hit this condition and never return the name of the runner?

Reply:
:coffee:   time ;) 

the if condition check if it's setup in redis, if it's the case we make a recursive call. line 108 
If the redis return None, the name is not set in the database so we can return it.  line 109

Reply:
I would see something like this:
```suggestion
        self.runner_counter += 1
        # Check that a virtual machine hasn't this name already
        runner =  redis_database.redis.get(f'runners:{name}')
        if runner:
            return runner['name']
        return name
```

Reply:
This function genreate a name therefore it must be an un used one.
WIth this we can have multiple VM having the same name

Reply:
aaaaaaaaaaaaaah now I see, I'm sorry. All good here

Reply:
I will add a explanation in the doc then